lib/iris/experimental/ugrid.py

```diff
@@ -204,6 +211,58 @@ def validate_arg_vs_list(arg_name, arg, valid_list):
             attributes=attributes,
         )
 
+    def __repr__(self):
+        def kwargs_filter(k, v):
+            result = False
+            if k != "cf_role":
+                if v is not None:
+                    result = True
+                    if (
+                        not isinstance(v, str)
+                        and isinstance(v, Iterable)
+                        and not v
+                    ):
+                        result = False
+                    elif k == "units" and v == "unknown":
+                        result = False
+            return result
+
+        def array2repr(array):
+            if self.has_lazy_indices():
+                result = repr(array)
```

Comment:
Note that, for the first time this will keep the payload lazy after performing a `repr(...)`.

As a follow-on pull-request, I'm going to apply this to all `_DimensionalMetadata` for `__repr__` and also make the `__str__` a slicker summary that also doesn't realise any lazy data... an obvious and small win :+1: